src/pages/index.js

```diff
@@ -0,0 +1,380 @@
+const previewModalCaptionElement =
+  previewModal.querySelector(".modal__caption");
+const previewModalCloseButton = previewModal.querySelector(".modal__close");
+
+//Card related elements
+const cardTemplate = document.querySelector("#card-template");
+const cardsList = document.querySelector(".cards__list");
+
+let selectedCard, selectedCardId;
+
+function handleDeleteSubmit(evt) {
+  console.log("deleting card with ID", selectedCardId);
+  evt.preventDefault();
+  setButtonText(modalDeleteButton, true, "Delete", "Deleting...");
+
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      setButtonText(modalDeleteButton, false, "Delete");
+
+      selectedCard.remove();
+      closeModal(deleteModal);
+    })
+    .catch(console.error);
+}
+
+function handleDeleteCard(evt, cardElement, cardId) {
+  evt.preventDefault();
+
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+
+  console.log(cardId);
+  openModal(deleteModal);
+}
+
+function handleLike(evt, id) {
+  // 1. Check if the card is currently liked or not
+  const isLiked = evt.target.classList.contains("card__like-button_liked");
+  console.log(id);
+  // 2. Call the changeLikeStatus method passing the appropriate arguments
+  api
+    .changeLikeStatus(id, !isLiked) // Pass the opposite of the current state
+    .then((data) => {
+      console.log(data);
+      // 3. Handle the response: toggle the 'liked' class based on the updated status
+      evt.target.classList.toggle("card__like-button_liked", data.isLiked);
+    })
+    .catch(console.error); // 4. Handle any errors in the catch block
+}
+
+function getCardElement(data) {
+  console.log(cardTemplate);
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+
+  const cardNameElement = cardElement.querySelector(".card__title");
+  const cardImageElement = cardElement.querySelector(".card__image");
+  const cardLikeButton = cardElement.querySelector(".card__like-button");
+  const deleteButton = cardElement.querySelector(".card__delete-button");
+
+  cardNameElement.textContent = data.name;
+  cardImageElement.src = data.link;
+  cardImageElement.alt = data.name;
+
+  if (data.isLiked) {
+    cardLikeButton.classList.add("card__like-button_liked");
+  }
+
+  cardLikeButton.addEventListener("click", () => {
+    console.log("Card ID before liking:", data._id);
+    cardLikeButton.classList.toggle("card__like-button_liked");
+  });
+
+  cardImageElement.addEventListener("click", () => {
+    previewModalImageElement.src = data.link;
+    previewModalCaptionElement.textContent = data.name;
+    previewModalImageElement.alt = data.name;
+    openModal(previewModal);
+  });
+  cardLikeButton.addEventListener("click", (evt) => handleLike(evt, data._id));
+
+  deleteButton.addEventListener("click", (evt) => {
+    handleDeleteCard(evt, cardElement, data._id);
+  });
+
+  modalCancelButton.addEventListener("click", () => {
```

Comment:
Please, remove it from `getCardElement`

The `cancel` button should be handled only **once** in the file body, otherwise you add listeners again and again to the same elements. This may cause **`memory leak`**